app/services/S3.scala

```diff
@@ -0,0 +1,94 @@
+package services
+
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.{Decoder, Encoder, Json, Printer}
+import io.circe.parser.decode
+import io.circe.syntax._
+import services.S3Client.RawVersionedS3Data
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+case class VersionedS3Data[T](value: T, version: String)
+
+trait S3Client {
+  def get(bucket: String, key: String)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]]
+  def put(bucket: String, key: String, data: RawVersionedS3Data)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]]
+}
+
+object S3Client {
+  type RawVersionedS3Data = VersionedS3Data[String]
+}
+
+object S3 extends S3Client with StrictLogging {
+
+  private val s3Client: AmazonS3 = AmazonS3ClientBuilder
+    .standard()
+    .withRegion(Aws.region)
+    .withCredentials(Aws.credentialsProvider)
+    .build()
+
+  def get(bucket: String, key: String)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]] = Future {
```

Comment:
wrapping a sync client in a future blocks an extra thread so can cause perf issues, this is a low traffic app so it won't cause anything in reality but it's better practice to use the async and complete a promise to avoid that.

Reply:
it seems the only way to do async s3 access is [TransferManager](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/services/s3/transfer/TransferManager.html), which looks a bit tedious.
Anyway, thread use is never going to be an issue for this app, but if it was I guess we could [configure the thread pool](https://www.playframework.com/documentation/2.6.x/ThreadPools#Highly-synchronous) differently